### PR TITLE
[react-events] Keyboard responder propagation handling

### PR DESCRIPTION
**Pull request by trueadm**
This adds support for conditional Keyboard responder propagation, for where there are nested keyboard responders. To allow a `onKeyDown` or `onKeyUp` event to propagate the callback is required to return `true` to propagate a single responder level. This is done by providing a new method on context called `continuePropagation` – which tells the responder propagation system to propagate a single level after processing the current responder (of the same responder type).

I also had to remove the usage of `invokeGuardedCallbackAndCatchFirstError` when calling the user-land event callback. This was because the existing `invokeGuardedCallbackAndCatchFirstError` doesn't support returning a value from the callback – which is what we need here. Attempting to change the mechanics of the function seemed far too invasive, so we can test this approach out internally to see if we run into any issues (we shouldn't though).

### CODE REVIEW

File: packages/legacy-events/ReactGenericBatching.js [hunk 1]
```diff
@@ -77,12 +75,12 @@ export function batchedEventUpdates(fn, a, b) {
...
 }
 
-export function executeUserEventHandler(fn: any => void, value: any) {
+// This is for the React Flare event system
+export function executeUserEventHandler(fn: any => void, value: any): any {
   const previouslyInEventHandler = isInsideEventHandler;
   try {
     isInsideEventHandler = true;
-    const type = typeof value === 'object' && value !== null ? value.type : '';
-    invokeGuardedCallbackAndCatchFirstError(type, fn, undefined, value);
```
Comment (gaearon) [inline packages/legacy-events/ReactGenericBatching.js]: Is there a follow-up to fix this? It seems important that event handler doesn't crash the rest of the code?

Comment (trueadm) [reply packages/legacy-events/ReactGenericBatching.js]: Yeah, I plan on addressing it this week.

Comment (necolas) [reply packages/legacy-events/ReactGenericBatching.js]: This was already fixed in a follow-up right? This guard is still there on master

Comment (trueadm) [reply packages/legacy-events/ReactGenericBatching.js]: LOL. I did, I’m clearly losing the plot.

**Pull request merged**